MovieDB/MovieDB/Scenes/Movie Detail/MovieDetailUseCase.swift

```diff
@@ -0,0 +1,21 @@
+import RxCocoa
+import RxSwift
+
+protocol MovieDetailUseCaseType {
+    func getMovieDetail(movieId: Int) -> Observable<MovieDetail>
+    func getMovieReviews(movieId: Int) -> Observable<[MovieReview]>
+}
+
+struct MovieDetailUseCase: MovieDetailUseCaseType {
+    private let repository = MovieDetailRepository()
+    
+    func getMovieDetail(movieId: Int) -> Observable<MovieDetail> {
+        let input = GetMovieDetailInput(movieId: movieId)
```

Comment:
input should be handled only inside Repository 
should pass  movieId to the Repository 
Only one command: return repository.getMovieDetail(movieId: movieId)